### PR TITLE
Update tests to increase branch coverage

### PR DESCRIPTION
**Pull request by abhishek-k-sharma-lab49**
Test case coverage increased to 96%

### CODE REVIEW

**Review by akhilesh-singh-Lab49 — APPROVED** [review]
looks good to me.

File: contracts/AbstractSwap.sol [hunk 1]
```diff
@@ -51,7 +51,7 @@ abstract contract AbstractSwap is HederaResponseCodes {
         pair.tokenB.tokenQty += _tokenBQty;
 
         associateToken(address(this),  _tokenA);
-        associateToken(address(this),  _tokenB);
```
Comment (amandeeprehal-lab49) [inline contracts/AbstractSwap.sol:54]: Pls remove this change

File: contracts/mock/MockBaseContract.sol [hunk 2]
```diff
@@ -5,22 +5,65 @@ import "../common/IBaseHTS.sol";
...
+        addLiquidityFailBTransfer,
+        removeLiquidityFailBTransfer,
+        removeLiquidity,
+        initialiseFailATransfer,
+        initialiseFailBTransfer
+    }
+
+    bool internal isSuccess;
+    int internal trueTransaction = 4;
+    FailTransactionFor internal failType;
```
Comment (amandeeprehal-lab49) [inline contracts/mock/MockBaseContract.sol:23]: we can create a public function that client can invoke to set the expectation.

Comment (amandeeprehal-lab49) [reply contracts/mock/MockBaseContract.sol:23]: e.g. setFailType function that takes enum as input parameter - 

`function setFailType(FailTransactionFor _failType) {
  failType = _failType;
}
`

then call this from unit test first as given then call business function like `addLiquidity` and then finally verify the expectation.

File: contracts/mock/MockBaseContract.sol [hunk 3]
```diff
@@ -5,22 +5,65 @@ import "../common/IBaseHTS.sol";
...
+            return 3;
         }
+        if (failType == FailTransactionFor.removeLiquidityFailBTransfer) {
+            return 1;
+        }
+        return 0;
+    }
+
+    function transferTokenPublic(address, address, address, int64) external override returns (int responseCode) {
+        if (trueTransaction + successForType() > 0 ) {
```
Comment (amandeeprehal-lab49) [inline contracts/mock/MockBaseContract.sol:49]: if we follow above then there is no need of this logic

File: contracts/mock/MockBaseContract.sol [hunk 4]
```diff
@@ -5,22 +5,65 @@ import "../common/IBaseHTS.sol";
...
+        }
+        int result = isSuccess ? int(22) : int(23);
+        return result;
+    }
     
-    function associateTokenPublic(address, address) external override view returns (int responseCode) {
-        return isSuccess ? int(22) : int(23);
+    function associateTokenPublic(address, address) external override returns (int responseCode) {
+        if (trueTransaction + successForType() > 0 ) {
+            trueTransaction-=1;
```
Comment (amandeeprehal-lab49) [inline contracts/mock/MockBaseContract.sol:59]: same here not required.

File: contracts/AbstractSwap.sol (hunk 1 above)
Comment (akhilesh-singh-Lab49) [reply contracts/AbstractSwap.sol:54]: Done

File: contracts/mock/MockBaseContract.sol (hunk 2 above)
Comment (akhilesh-singh-Lab49) [reply contracts/mock/MockBaseContract.sol:25]: Done

File: contracts/mock/MockBaseContract.sol (hunk 3 above)
Comment (akhilesh-singh-Lab49) [reply contracts/mock/MockBaseContract.sol]: Done

File: contracts/mock/MockBaseContract.sol (hunk 4 above)
Comment (akhilesh-singh-Lab49) [reply contracts/mock/MockBaseContract.sol:73]: Done

File: contracts/mock/MockBaseContract.sol [hunk 5]
```diff
@@ -3,19 +3,76 @@ pragma solidity ^0.8.0;
...
     }
-    function transferTokenPublic(address, address, address, int64) 
-        external view override
-returns (int responseCode) {
-            return isSuccess ? int(22) : int(23);
+
+    function setFailType(int _type) public {
+        failType = FailTransactionFor(_type);
+        trueTransaction = successForType();
+
```
Comment (amandeeprehal-lab49) [inline contracts/mock/MockBaseContract.sol:33]: why extra space?

File: contracts/AbstractSwap.sol [hunk 6]
```diff
@@ -51,7 +51,7 @@ abstract contract AbstractSwap is HederaResponseCodes {
...
+        associateToken(address(this),  _tokenB);
```
Comment (amandeeprehal-lab49) [inline contracts/AbstractSwap.sol:54]: This file should not show in difference, it looks its still there.

**Pull request merged**